src/main/scala/nl/knaw/dans/easy/stage/dataset/Util.scala

```diff
@@ -67,6 +67,25 @@ object Util {
     } yield audience.text
   }
 
+  def readFileType(filePath: String)(implicit s: Settings): Try[Option[String]] = Try {
+    val filetype = for {
+      file <- loadBagXML("metadata/files.xml") \\ "files" \ "file"
+      if (file \ "@filepath").text == filePath
+      filetype <- file \ "type"
+    } yield filetype
+    if (filetype.size == 1) Option(filetype.head.text)
+    else None
```

Comment:
Why shouldn't optimization be an issue? What if you have a `files.xml` with a billion files in it and the first one is the one you're looking for. Why do you want to keep looking through all the others? That's just wasting time! Therefore I neither see why you would return a `Seq[Node]` instead.